build/codegen/codegen.sh

```diff
@@ -14,29 +14,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-GROUP_VERSIONS="ceph.rook.io:v1"
+##GROUP_VERSIONS="ceph.rook.io:v1"
+
+scriptdir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+basedir="$(cd "${scriptdir}/../.."  && pwd)"
+apisdir="${basedir}"/pkg/apis
+clientdir="${basedir}"/pkg/client
+
+boilerplate="${CODE_GENERATOR}/examples/hack/"boilerplate.go.txt
+kube_codegen="${CODE_GENERATOR}/kube_codegen.sh"
+
+
+echo "DEBUG: scriptdir: ${scriptdir}"
+echo "DEBUG: basedir: ${basedir1}"
+echo "DEBUG: apisdir: ${apisdir}"
+echo "DEBUG: clientdir: ${clientdir}"
+echo "DEBUG: CODE_GENERATOR: ${CODE_GENERATOR}"
+echo "DEBUG: boilerplate ${boilerplate}"
+echo "DEBUG: kube_codegen: ${kube_codegen}"
+ echo "DEBUG: done"
+# exit 0
+
+
+
+set -o pipefail
+
+
+source "${kube_codegen}"
+
+
 
-scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 # CODE GENERATION
 # we run deepcopy and client,lister,informer generations separately so we can use the flag "--plural-exceptions"
 # which is only known by client,lister,informer binary and not the deepcopy binary
+#
+
+
```

Comment:
Please get rid of double empty lines before final revision.

Reply:
done (at least some.